Tasks/AzureCloudPowerShellDeployment/Publish-AzureCloudDeployment-Legacy.ps1

```diff
@@ -0,0 +1,296 @@
+                            return
+                        }
+                    }
+                    else
+                    {
+                        #If we don't find a StorageAccount in the XML file, use the one associated with the definition's storage account
+                        Write-Verbose "No StorageAccount found in PublicConfig.  Using the storage account set on the definition..."
+                        $storageContext = $definitionStorageContext
+                    }
+
+                    Write-Host "New-AzureServiceDiagnosticsExtensionConfig -Role $role -StorageContext <context> -DiagnosticsConfigurationPath $fullExtPath"
+                    $wadconfig = New-AzureServiceDiagnosticsExtensionConfig -Role $role -StorageContext $storageContext -DiagnosticsConfigurationPath $fullExtPath 
+
+                    #Add each extension configuration to the array for use by caller
+                    $diagnosticsConfigurations += $wadconfig
+                }
+            }
+        }
+        else
+        {
+            Write-Warning (Get-LocalizedString -Key "Could not get the primary storage key for storage account '{0}'. Unable to apply any diagnostics extensions." -ArgumentList "$storageAccount")
+        }
+    }
+    
+    return $diagnosticsConfigurations
+}
+
+Write-Verbose "Entering script Publish-AzureCloudDeployment.ps1"
+
+import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
+import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
+
+Write-Host "ConnectedServiceName= $ConnectedServiceName "
+Write-Host "ServiceName= $ServiceName"
+Write-Host "ServiceLocation= $ServiceLocation"
+Write-Host "AffinityGroup= $AffinityGroup"
+Write-Host "StorageAccount= $StorageAccount"
+Write-Host "CsPkg= $CsPkg"
+Write-Host "CsCfg= $CsCfg"
+Write-Host "Slot= $Slot"
+Write-Host "DeploymentLabel= $DeploymentLabel"
+Write-Host "AppendDateTimeToLabel= $AppendDateTimeToLabel"
+Write-Host "AllowUpgrade= $AllowUpgrade"
+Write-Host "NewServiceAdditionalArguments= $NewServiceAdditionalArguments"
+
+$allowUpgrade = Convert-String $AllowUpgrade Boolean
+
+Write-Host "Find-Files -SearchPattern $CsCfg"
+$serviceConfigFile = Find-Files -SearchPattern "$CsCfg"
+Write-Host "serviceConfigFile= $serviceConfigFile"
+$serviceConfigFile = Get-SingleFile $serviceConfigFile $CsCfg
+
+Write-Host "Find-Files -SearchPattern $CsPkg"
+$servicePackageFile = Find-Files -SearchPattern "$CsPkg"
+Write-Host "servicePackageFile= $servicePackageFile"
+$servicePackageFile = Get-SingleFile $servicePackageFile $CsPkg
+
+Write-Host "Get-AzureService -ServiceName $ServiceName -ErrorAction SilentlyContinue  -ErrorVariable azureServiceError"
+$azureService = Get-AzureService -ServiceName $ServiceName -ErrorAction SilentlyContinue  -ErrorVariable azureServiceError
+
+if($azureServiceError){
+   $azureServiceError | ForEach-Object { Write-Warning $_.Exception.ToString() }
```

Comment:
It might be a better customer experience to write the message only to the warning pipeline and then write the exception.tostring to the verbose pipeline.